src/Symfony/Bundle/FrameworkBundle/CacheWarmer/ValidatorCacheWarmer.php

```diff
@@ -77,14 +78,14 @@ protected function warmUpPhpArrayAdapter(PhpArrayAdapter $phpArrayAdapter, array
     /**
      * @param LoaderInterface[] $loaders
      *
-     * @return XmlFileLoader[]|YamlFileLoader[]
+     * @return list<XmlFileLoader|YamlFileLoader|AttributeLoader>
      */
     private function extractSupportedLoaders(array $loaders): array
     {
         $supportedLoaders = [];
 
         foreach ($loaders as $loader) {
-            if ($loader instanceof XmlFileLoader || $loader instanceof YamlFileLoader) {
+            if (method_exists($loader, 'getMappedClasses')) {
```

Comment:
This method existence should be checked using an interface. For compatibility with older versions of the validator component, we can keep checking the concrete classes.
```suggestion
            if ($loader instanceof XmlFileLoader || $loader instanceof YamlFileLoader || $loader instanceof WarmableLoaderInterface)
```